crates/turborepo-lib/src/commands/link.rs

```diff
@@ -0,0 +1,206 @@
+use std::{
+    fs,
+    fs::{File, OpenOptions},
+    io,
+    io::{BufRead, Write},
+    os::unix::fs::PermissionsExt,
+};
+
+use anyhow::{anyhow, Context, Result};
+use console::Style;
+use dialoguer::{theme::ColorfulTheme, Confirm, Select};
+use dirs_next::home_dir;
+
+use crate::{
+    client::{CachingStatus, Team, UserClient},
+    commands::CommandBase,
+    ui::{BOLD, CYAN, GREY, UNDERLINE},
+};
+
+enum SelectedTeam<'a> {
+    User,
+    Team(&'a Team),
+}
+
+pub async fn link(mut base: CommandBase, modify_gitignore: bool) -> Result<()> {
+    let homedir_path = home_dir().ok_or_else(|| anyhow!("could not find home directory."))?;
+    let homedir = homedir_path.to_string_lossy();
+    println!(">>> Remote Caching");
```

Comment:
Once again going to ask to combine these `println` statements to avoid locking/unlocking stdout unnecessarily